### DISCUSSION TITLE
How to attach restored volume to Pod

### DISCUSSION THREAD

**Discussion by zainul1114** [longhorn/longhorn#6346] [General]
## Question
> Suggest to use https://github.com/longhorn/longhorn/discussions to ask questions.

## Environment

 - Longhorn version: v1.4.2
 - Kubernetes version: 1.22.3
 - Node config 1master+1node
   - OS type and version: RHEL 7.9
   - CPU per node: 4
   - Memory per node: 8
   - Disk type: DAS
   - Network bandwidth and latency between the nodes: 
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal

## Additional context

My Scenario is as below, have 2nodes - kubem and kuben
![image](https://github.com/longhorn/longhorn/assets/31759122/8e12c0dc-214c-4033-9858-d3ed4e43f64a)

1. I have created a portainer pod using portainer.yaml 
2. Once created i have taken backup on longhorn and I have deleted the volume initially created one.
![image](https://github.com/longhorn/longhorn/assets/31759122/da406d96-31ef-4d11-862c-68c2bc13ca25)

3. restored volume
![image](https://github.com/longhorn/longhorn/assets/31759122/a2109d78-6bec-4b3c-9eaa-a7ef140001ec)

4. but i am not able to attach volume to pod

![image](https://github.com/longhorn/longhorn/assets/31759122/6545981e-34e9-4c06-9816-3efc9aa8c23b)

Need your inupts, can we attach the volume to pod.

Regards,
Zain

**Comment by mantissahz** [comment]
I think this might help https://longhorn.io/docs/1.5.0/snapshots-and-backups/backup-and-restore/restore-statefulset/
or You can create a PVC manually for this restored volume to apply it in the manifest https://longhorn.io/docs/1.5.0/volumes-and-nodes/create-volumes/#pvpvc-creation-for-existing-longhorn-volume

**Reply by zainul1114** [reply]
Hi Team, i have assigned the volume to pod but pod status is pending state. 

![image](https://github.com/longhorn/longhorn/assets/31759122/9fadd95f-a4ef-4cac-a465-bda5f051d37f)

![image](https://github.com/longhorn/longhorn/assets/31759122/0882752e-140d-4d7d-87d2-95d94deaf535)

![image](https://github.com/longhorn/longhorn/assets/31759122/33c3ba3c-349f-4fe7-959f-482b5ad0236f)

Please help on this.

**Reply by mantissahz** [reply]
You might change the `claimName` of `persistentVolumeClaim` in `volumes` in the manifest.
I believe the PVC name is `pvc-ffd9aea8-457b-8db5-xxxx`, right?
```txt
Volumes:
  data:
    Type: PersistentVolumeClaim
    ClaimName: portainer
```
